File: 0062-unique-paths/0062-unique-paths.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int uniquePaths(int m, int n) {
        // Create a 2D DP table with dimensions m x n
        vector<vector<int>> dp(m, vector<int>(n, 1));  // Initialize first row and first column to 1
        
        // Fill the DP table
        for (int i = 1; i < m; ++i) {
            for (int j = 1; j < n; ++j) {
                dp[i][j] = dp[i - 1][j] + dp[i][j - 1];  // Sum of ways from above and left
            }
        }
        
        return dp[m - 1][n - 1];  // The bottom-right corner contains the result
    }
};

File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[1] * n for _ in range(m)]  # Initialize DP table with 1s

        for i in range(1, m):
            for j in range(1, n):
                dp[i][j] = dp[i - 1][j] + dp[i][j - 1]

        return dp[m - 1][n - 1]